sdk/pinocchio/src/account_info.rs

```diff
@@ -834,4 +881,44 @@ mod tests {
         assert_eq!(lamports, 200);
         assert_eq!(state, 0);
     }
+
+    #[test]
+    fn test_map_split() {
+        // Create a mutable array and a dummy borrow state
+        let mut data: [u8; 4] = [1, 2, 3, 4];
+        let mut state: u8 = 0b_0000_1000; // Simulate a mutable borrow
+        use core::ptr::NonNull;
+        use core::marker::PhantomData;
+
+        // Create a RefMut<[u8]>
+        let ref_mut = RefMut {
+            value: NonNull::from(&mut data[..]),
+            state: NonNull::from(&mut state),
+            borrow_mask: DATA_MASK,
+            marker: PhantomData,
+        };
+
+        // Split the RefMut into two non-overlapping slices
+        let (mut left, mut right) = RefMut::map_split(ref_mut, |slice| slice.split_at_mut(2));
+
+        // Mutate both halves independently
+        left.copy_from_slice(&[10, 20]);
+        right.copy_from_slice(&[30, 40]);
+
+        // assert the changes
+        assert_eq!(&data, &[10, 20, 30, 40]);
+
+        // drop one half and ensure the other still works
+        drop(left);
+        // right is still valid
```

Comment:
Unfortunately, after the first drop, the mutable flag gets reset since `left` and `right` chare the same borrow state. You would expect that adding this:
```rust
// borrow state should still indicate mutable borrow
assert_eq!(state & 0b_0000_1000, 0b_0000_1000);
```
to be valid since `right` is still borrowed, but this is not the case.